lib/forcex/auth/session_id.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Forcex.Auth.SessionId do
+  @moduledoc """
+  Auth via a session id
+  """
+
+  @behaviour Forcex.Auth
+
+  def login(conf, starting_struct) do
+    schema = "http://www.w3.org/2001/XMLSchema"
+    schema_instance = "http://www.w3.org/2001/XMLSchema-instance"
+    env = "http://schemas.xmlsoap.org/soap/envelope/"
+
+    envelope = """
+    <?xml version="1.0" encoding="utf-8" ?>
+    <env:Envelope xmlns:xsd="#{schema}" xmlns:xsi="#{schema_instance}" xmlns:env="#{env}">
+    <env:Body>
+    <n1:login xmlns:n1="urn:partner.soap.sforce.com">
+      <n1:username>#{conf.username}</n1:username>
+      <n1:password>#{conf.password}#{conf.security_token}</n1:password>
+    </n1:login>
+    </env:Body>
+    </env:Envelope>
+    """
+
+    headers = [
+      {"Content-Type", "text/xml; charset=UTF-8"},
+      {"SOAPAction", "login"}
+    ]
+
+    "https://login.salesforce.com/services/Soap/u/#{starting_struct.api_version}"
+    |> HTTPoison.post!(envelope, headers)
```

Comment:
@mmrobins it looks like we have different semantics for `login` between the 2 implementations of the `Forcex.Auth` behaviour. This one will raise if it encounters a problem whereas the other will return an error tuple (I think. I might have to dig in further). Regardless, it would likely be beneficial to have `login` and `login!`. It doesn't need to be part of this PR though.

Reply:
I dug in. They're all `!` functions underneath, despite outward appear. Present-Jeff is shaming Past-Jeff.

You're good to go with this PR.